src/ray/core_worker/core_worker.cc

```diff
@@ -495,8 +474,7 @@ Status CoreWorker::Contains(const ObjectID &object_id, bool *has_object) {
   if (!found) {
     // We check plasma as a fallback in all cases, since a direct call object
     // may have been spilled to plasma.
-    RAY_RETURN_NOT_OK(plasma_store_provider_->Contains(
-        object_id.WithTransportType(TaskTransportType::RAYLET), &found));
+    RAY_RETURN_NOT_OK(plasma_store_provider_->Contains(object_id, &found));
```

Comment:
This should be ok. There can be an optimization for memory store to return the object ids that are with InPlasmaError.

Reply:
I changed it to return `in_plasma`, and we only check plasma then